fastai/data/block.py

```diff
@@ -107,25 +115,41 @@ def __init__(self, blocks=None, dl_type=None, getters=None, n_inp=None, item_tfm
     def _combine_type_tfms(self): return L([self.getters, self.type_tfms]).map_zip(
         lambda g,tt: (g.fs if isinstance(g, Pipeline) else L(g)) + tt)
 
-    def new(self, item_tfms=None, batch_tfms=None):
+    def new(self,
+        item_tfms:list=None,  # List of `ItemTransform`s, applied on an item
+        batch_tfms:list=None, # List of `Transform`s or `RandTransform`s, applied by batch
+    ):
         self.item_tfms  = _merge_tfms(self.default_item_tfms,  item_tfms)
         self.batch_tfms = _merge_tfms(self.default_batch_tfms, batch_tfms)
         return self
 
     @classmethod
-    def from_columns(cls, blocks=None, getters=None, get_items=None, **kwargs):
+    def from_columns(cls,
+        blocks:list =None, # List of `TransformBlock`s
+        getters:list =None, # List of getter functions
+        get_items:callable=None, # A function to get items
+        **kwargs,
+    ):
         if getters is None: getters = L(ItemGetter(i) for i in range(2 if blocks is None else len(L(blocks))))
         get_items = _zip if get_items is None else compose(get_items, _zip)
         return cls(blocks=blocks, getters=getters, get_items=get_items, **kwargs)
 
-    def datasets(self, source, verbose=False):
+    def datasets(self,
+        source, # The data source
+        verbose:bool=False, # Show verbose messages
+    ):
         self.source = source                     ; pv(f"Collecting items from {source}", verbose)
         items = (self.get_items or noop)(source) ; pv(f"Found {len(items)} items", verbose)
         splits = (self.splitter or RandomSplitter())(items)
         pv(f"{len(splits)} datasets of sizes {','.join([str(len(s)) for s in splits])}", verbose)
         return Datasets(items, tfms=self._combine_type_tfms(), splits=splits, dl_type=self.dl_type, n_inp=self.n_inp, verbose=verbose)
 
-    def dataloaders(self, source, path='.', verbose=False, **kwargs):
+    def dataloaders(self,
+        source,  # The data source
+        path:str='.', # The path to the data source
+        verbose:bool=False, # Show verbose messages
+        **kwargs
+    ):
```

Comment:
`path` also has a sneaky feature where it sets the `Learner` path location too
```suggestion
    def dataloaders(self,
        source,  # The data source
        path:str='.', # Data source and default `Learner` path
        verbose:bool=False, # Show verbose messages
        **kwargs
    ):
```